structs.go

```diff
@@ -45,21 +45,50 @@ const (
 )
 
 type User struct {
-	ID       string `json:"_id"`
-	Username string `json:"username"`
-	Name     string `json:"name"`
+	ID           string `json:"_id"`
+	Username     string `json:"username"`
+	Name         string `json:"name"`
+	Token        string `json:"token,omitempty"`
+	TokenExpires int64  `json:"tokenExpires,omitempty"`
 }
 
 type Message struct {
-	ID           string   `json:"_id"`
-	Type         string   `json:"t,omitempty"`
-	CreationDate JsonDate `json:"ts,omitempty"`
-	Message      string   `json:"msg"`
-	Url          []string `json:"url,omitempty"`
-	ExpireAt     JsonDate `json:"url,omitempty"`
-	Mentions     []string `json:"mentions,omitempty"`
-	Sender       *User    `json:"u,omitempty"`
-	ChannelID    string   `json:"rid,omitempty"`
+	ChannelID string `json:"rid"`
+	Message   string `json:"msg"`
+
+	// Optional fields, or for receiver only
+	ID          string       `json:"_id,omitempty"`
+	Timestamp   *JsonDate    `json:"ts,omitempty"`
+	Sender      *User        `json:"u,omitempty"`
+	Alias       string       `json:"alias,omitempty"`
+	Avatar      string       `json:"avatar,omitempty"`
+	Attachments []Attachment `json:"attachments,omitempty"`
+	Emoji       string       `json:"emoji,omitempty"`
+}
```

Comment:
https://rocket.chat/docs/developer-guides/realtime-api/the-message-object/ && https://rocket.chat/docs/developer-guides/rest-api/chat/postmessage/